### PR TITLE
CI: Initial code coverage configuration with Codecov

### PR DESCRIPTION
**Pull request by echoix**
This PR is the first step enabling code coverage in the grass repo.

It will require authorizing the Codecov app for the repo, and then adding the CODECOV_TOKEN to the repository secrets.
Follow the steps here: https://docs.codecov.com/docs/quick-start

For this first iteration, it enables code coverage only for the simplest code coverage step. In python only, through pytest, without any multiple test workers running. I made sure that a no-code change didn't make a variation in the covered lines.

### CODE REVIEW

**Comment by echoix** [comment]
@neteler I made a request to authorize the Codecov application to the repo, but Vacláv wasn't admin enough to approve it by himself. Could you take a look into it, and add the token to the repository secrets as detailed in the guide linked in the description?

**Comment by neteler** [comment]
I have tried to activate it but get stuck here:

image

The continue button doesn't do anything. Will try with a different browser.

**Comment by neteler** [comment]
It seems to do something:

https://app.codecov.io/github/OSGeo/grass/tree/echoix%2Fgrass%3Ainitial-pytest-codecov-configuration

image

**Comment by echoix** [comment]
So that means that the configuration is nearly complete. I'll merge main into the PR to really test the upload, as I'm surprised the coverage info is already there (probably from my branch)

**Comment by echoix** [comment]
So, the upload worked, and appears correctly, the change from the merge did something, so that means that this enablement PR is ready. Future PRs will be able to compare what changed or not, and upload other reports from different code coverage environments (c/c++ from gcov, unittest ones, pytest with parallel tests, macOS, etc.). Having this will enable to confirm each of the testing infrastructure change is equivalent or better.

For this on-boarding period, the checks are non failing (informational: true), no comments are posted, etc. It is the bare bones until we get used to it.

**Comment by neteler** [comment]
> It will require authorizing the Codecov app for the repo, and then adding the CODECOV_TOKEN to the repository secrets.

I have now added the CODECOV_TOKEN under https://github.com/OSGeo/grass/settings/secrets/actions

**Comment by echoix** [comment]
So it is done now

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    - component_id: notebooks
+      name: notebooks (jupyter)
+      paths:
+        - python/grass/jupyter/**
+        - "**/*.ipynb"
+    - component_id: python_library
+      name: python library
+      paths:
+        - python/grass/**
+# When modifying this file, please validate using
```
Comment (neteler) [inline codecov.yml:111]: minor idea: move this comment to the top for better visibility?

Comment (echoix) [reply codecov.yml:111]: The template/examples from the codecov docs that I started from had it below. I can bring it up if you think it's better.

Comment (echoix) [reply codecov.yml:111]: Should I?

Comment (echoix) [reply codecov.yml]: Done the change either way

**Comment by echoix** [comment]
I'd like to get this through the finish line. Is there any else I have do to help it?

**Review by neteler — APPROVED** [review]
Thanks for your work!

**Comment by echoix** [comment]
I'll have next parts in the next days of the weekend (hopefully), as now I'll be able to rely on this to confirm that the changes are equivalent (at least with respect to what is currently covered)

**Pull request merged**